### PR TITLE
Follow up on serving different home page

### PR DESCRIPTION
**Pull request by bjaxelsen**
I've made an alter of Drupal core's route handling thrugh implementation of \Drupal\Core\DependencyInjection\ServiceProviderBase::alter() and made changes to the test of delivering different home pages.

### CODE REVIEW

**Comment by bjaxelsen** [comment]
See https://www.drupal.org/node/2662196 and #166

File: domain_config/src/DomainConfigServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Overrides the router.route_provider service to point to our customized one.
+ */
+class DomainConfigServiceProvider extends ServiceProviderBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function alter(ContainerBuilder $container) {
+    $definition = $container->getDefinition('router.route_provider');
+    $definition->setClass('Drupal\domain_config\Routing\DomainRouteProvider');
```
Comment (Crell) [inline domain_config/src/DomainConfigServiceProvider.php]: In PHP 5.5, it's safe and I would argue preferred to use the class-constant-form for a class name.  That is:

``` php
setClass(DomainRouteProvider::class);
```

And then also add a "use" statement above for that class. That way your IDE can recognize it as a class name, spell check it, find it in refactor commands, etc. etc.

Comment (bjaxelsen) [inline domain_config/src/DomainConfigServiceProvider.php]: thanks @Crell - I did not know this before now :-)

File: domain_config/src/Tests/DomainConfigHomepageTest.php [hunk 2]
```diff
@@ -48,10 +50,22 @@ function testDomainConfigHomepage() {
     ));
     $homepages = $this->getHomepages();
     foreach ($domains as $domain) {
+      // Clone the site configuration to set individual home pages if different
+      // than default
+      if ($homepages[$domain->id()] != 'node') {
```
Comment (agentrickard) [inline domain_config/src/Tests/DomainConfigHomepageTest.php]: These should be picked up from `/tests/modules/domain_config_test/config/install`. Is that not happening?

**Comment by agentrickard** [comment]
That Travis failure looks like network interruption. Re-running tests.

I like this approach, as it bypasses the need for a core patch.

**Comment by bjaxelsen** [comment]
@agentrickard I had not seen the yml files in the test module, so no need to set the domain homepages again in the DomainConfigHomepageTest - I've cleaned this up.

**Comment by agentrickard** [comment]
Great. Thank you.

**Pull request merged**